sdk/attestation/Azure.Security.Attestation/src/Models/AttestationSigner.cs

```diff
@@ -19,9 +19,9 @@ public class AttestationSigner
         /// </summary>
         /// <param name="signingCertificates"></param>
         /// <param name="certificateKeyId"></param>
-        public AttestationSigner(X509Certificate2[] signingCertificates, string certificateKeyId)
+        public AttestationSigner(IEnumerable<X509Certificate2> signingCertificates, string certificateKeyId)
         {
-            SigningCertificates = signingCertificates;
+            SigningCertificates = (IReadOnlyList<X509Certificate2>)signingCertificates;
             CertificateKeyId = certificateKeyId;
         }
 
```

Comment:
On L45,  we're doing a `ToArray()` call on a list in order to return a `ReadOnlyList`, which is going to result in allocating an array and coping the data.    We may want to consider changing that to `returnedCertificates.AsReadOnly()` instead.